### PR TITLE
Notify users that they have unminted Raha.

### PR DESCRIPTION
**Pull request by rahulgi**

### CODE REVIEW

File: packages/server/src/helpers/calculateMaxMintableForMember.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { DocumentSnapshot } from "@google-cloud/firestore";
+import Big from "big.js";
+
+const RAHA_UBI_WEEKLY_RATE = 10;
```
Comment (tinaroh) [inline packages/server/src/helpers/calculateMaxMintableForMember.ts]: not necessary right now, but maybe worth putting RAHA_UBI_WEEKLY_RATE in shared

File: packages/server/src/routes/cron/notifyOnUnminted.ts [hunk 2]
```diff
@@ -1,29 +1,134 @@
...
+ * EH (Western Sahara), and TA (Tristan da Cunha). Good enough IMO.
+ */
+function getDefaultOffsetForPhoneNumber(
+  phoneNumber: string
+): string | undefined {
+  const phoneNumberUtil = PhoneNumberUtil.getInstance();
+  const countryCode = phoneNumberUtil.parse(phoneNumber).getCountryCode();
+  if (countryCode) {
+    const regionCode = phoneNumberUtil.getRegionCodeForCountryCode(countryCode);
+    const defaultTimezoneByCountryCode = require("default-timezone-for-country/data/default_timezone_by_country_code.json");
```
Comment (tinaroh) [inline packages/server/src/routes/cron/notifyOnUnminted.ts]: does this load the file every time the function is called? or does it keep it around?

File: packages/server/src/routes/cron/notifyOnUnminted.ts [hunk 3]
```diff
@@ -1,29 +1,134 @@
...
+    if (timezone) {
+      return timezone.utc_offset;
+    }
+  }
+  return undefined;
+}
+
+/**
+ * Return that we should notify the user if all the above is true:
+ * * their max-mintable is >= the mint camp.
```
Comment (tinaroh) [inline packages/server/src/routes/cron/notifyOnUnminted.ts]: nit: sp camp -> cap

File: packages/server/src/routes/cron/notifyOnUnminted.ts [hunk 4]
```diff
@@ -1,29 +1,134 @@
...
+import { createApiRoute } from "..";
 import { sendPushNotification } from "../../helpers/sendPushNotification";
-import { messaging as adminMessaging } from "firebase-admin";
+import {
+  calculateMaxMintableForMember,
+  MINT_CAP
+} from "../../helpers/calculateMaxMintableForMember";
+
+// Notify people if it is 12 o'clock in their local time.
+const HOUR_TO_NOTIFY = 12;
```
Comment (tinaroh) [inline packages/server/src/routes/cron/notifyOnUnminted.ts]: https://momentjs.com/docs/#/get-set/hour/ I think hour is 0 indexed

File: packages/server/src/routes/cron/notifyOnUnminted.ts [hunk 5]
```diff
@@ -1,29 +1,134 @@
...
+  const maxMintable = calculateMaxMintableForMember(member);
+  if (maxMintable.gte(MINT_CAP)) {
+    const userRecord = await auth.getUser(member.id);
+    const phoneNumber = userRecord.phoneNumber;
+    if (phoneNumber) {
+      const offset = getDefaultOffsetForPhoneNumber(phoneNumber);
+      const localTime = moment().utcOffset(offset || 0);
+      if (localTime.hour() === HOUR_TO_NOTIFY) {
+        if (memberNotificationHistory.exists) {
+          return !memberNotificationHistory.get("notifiedOnUnminted");
```
Comment (tinaroh) [inline packages/server/src/routes/cron/notifyOnUnminted.ts]: nit: Should we name it something like `notifiedOnUnmintedOverCap`? I imagine unminted could also mean the regularly scheduled weekly unminted notice

File: packages/server/src/routes/cron/notifyOnUnminted.ts [hunk 6]
```diff
@@ -1,29 +1,134 @@
...
+          const member = await transaction.get(membersCollection.doc(m.id));
+          const memberNotificationHistory = await transaction.get(
+            notificationHistoryCollection.doc(member.id)
+          );
+          if (await shouldNotify(auth, member, memberNotificationHistory)) {
+            const title = `Mint ${calculateMaxMintableForMember(member)
+              .round(2, 0)
+              .toString()} Raha before it's too late!`;
+            const body =
+              "Your mintable amount will be capped at 40 Raha on November 15th. Tap this notification to mint now and avoid losing Raha!";
```
Comment (tinaroh) [inline packages/server/src/routes/cron/notifyOnUnminted.ts]: 11/15 or Nov. 15?

File: packages/server/src/helpers/calculateMaxMintableForMember.ts (hunk 1 above)
Comment (rahulgi) [reply packages/server/src/helpers/calculateMaxMintableForMember.ts]: agreed, I'll leave a TODO for this.

File: packages/server/src/routes/cron/notifyOnUnminted.ts (hunk 2 above)
Comment (rahulgi) [reply packages/server/src/routes/cron/notifyOnUnminted.ts]: supposedly they're cached: https://stackoverflow.com/questions/9210542/node-js-require-cache-possible-to-invalidate

File: packages/server/src/routes/cron/notifyOnUnminted.ts (hunk 3 above)
Comment (rahulgi) [reply packages/server/src/routes/cron/notifyOnUnminted.ts]: oops ty lol

File: packages/server/src/routes/cron/notifyOnUnminted.ts (hunk 6 above)
Comment (rahulgi) [reply packages/server/src/routes/cron/notifyOnUnminted.ts]: hmm I guess 11/15 to be consistent

File: packages/server/src/routes/cron/notifyOnUnminted.ts (hunk 4 above)
Comment (rahulgi) [reply packages/server/src/routes/cron/notifyOnUnminted.ts]: 12 o'clock is already zero indexed, since midnight is 0-o'clock. the one weird place where we zero-index in everyday life lol

File: packages/server/src/routes/cron/notifyOnUnminted.ts (hunk 5 above)
Comment (rahulgi) [reply packages/server/src/routes/cron/notifyOnUnminted.ts]: sgtm

File: packages/server/src/routes/cron/notifyOnUnminted.ts (hunk 6 above)
Comment (rahulgi) [reply packages/server/src/routes/cron/notifyOnUnminted.ts]: Although, I think I prefer Nov 15. I find 11/15 hard to parse. I'll go with that.
